src/dolphin/atmosphere/ionosphere.py

```diff
@@ -0,0 +1,388 @@
+    freq: float
+        radar carrier frequency in Hz.
+    obs_type: str
+        given the same iono, the impact on offset (amplitude) and phase is reversed.
+
+    Returns
+    -------
+    np.ndarray
+        predicted range delay in meters
+    """
+    # ignore no-data value in inc_angle
+    if isinstance(inc_angle, np.ndarray):
+        inc_angle[inc_angle == 0] = np.nan
+
+    # convert to TEC in LOS based on equation (3) in Chen and Zebker (2012)
+
+    # Equation (26) in Bohm & Schuh (2013) Chapter 2.
+    n_iono_group = 1 + K * vtec * 1e16 / freq**2
+
+    # Equation (8) in Yunjun et al. (2022, TGRS)
+    ref_angle = (
+        np.arcsin(1 / n_iono_group * np.sin(inc_angle * np.pi / 180)) * 180 / np.pi
+    )
+
+    tec = vtec / np.cos(ref_angle * np.pi / 180.0)
+
+    # calculate range delay based on equation (1) in Chen and Zebker (2012)
+    range_delay = (tec * 1e16 * K / (freq**2)).astype(np.float32)
+
+    # group delay = phase advance * -1
+    if obs_type != "phase":
+        range_delay *= -1.0
+
+    return range_delay
+
+
+def get_ionex_value(
+    tec_file: Filename, utc_sec: float, lat: float, lon: float
+) -> float:
+    """Get the TEC value from input IONEX file for the input lat/lon/datetime.
+
+    Parameters
+    ----------
+    tec_file: Filename
+        path of local TEC file
+    utc_sec: float
+        UTC time of the day in seconds
+    lat: float
+        latitude in degrees
+    lon: float
+        longitude in degrees
+
+
+    Returns
+    -------
+    float
+        vertical TEC value in TECU
+    """
+    # time info
+    utc_min = utc_sec / 60.0
+
+    # read TEC file
+    mins, lats, lons, tec_maps = read_ionex(tec_file)
+
+    # interpolate between consecutive rotated TEC maps
+    # reference: equation (3) in Schaer et al. (1998)
+
+    ind0 = np.where((mins - utc_min) <= 0)[0][-1]
+    ind1 = ind0 + 1
+
+    lon0 = lon + (utc_min - mins[ind0]) * 360.0 / (24.0 * 60.0)
+    lon1 = lon + (utc_min - mins[ind1]) * 360.0 / (24.0 * 60.0)
+
+    tec_val0 = interpolate.griddata(
+        points=(lons.flatten(), lats.flatten()),
+        values=tec_maps[ind0, :, :].flatten(),
+        xi=(lon0, lat),
+        method="linear",
+    )
+
+    tec_val1 = interpolate.griddata(
+        points=(lons.flatten(), lats.flatten()),
+        values=tec_maps[ind1, :, :].flatten(),
+        xi=(lon1, lat),
+        method="linear",
+    )
+
+    tec_val = (mins[ind1] - utc_min) / (mins[ind1] - mins[ind0]) * tec_val0
+    +(utc_min - mins[ind0]) / (mins[ind1] - mins[ind0]) * tec_val1
+
+    return tec_val
+
+
+def read_ionex(
+    tec_file: Filename,
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """Read TEC file in IONEX format.
+
+    Parameters
+    ----------
+    tec_file: Filename
+        path to the TEC file in IONEX format
+
+    Returns
+    -------
+    mins: np.ndarray
+        1D np.ndarray in size of (num_map), time of the day in minutes
```

Comment:
I see that `num_map` is in the ionex file, but can we put some explantion in the docstring of what that means? I actually am not quite sure what they mean with "number of maps in the file"